patches/minecraft/net/minecraft/client/renderer/tileentity/SignTileEntityRenderer.java.patch

```diff
@@ -0,0 +1,23 @@
+--- a/net/minecraft/client/renderer/tileentity/SignTileEntityRenderer.java
++++ b/net/minecraft/client/renderer/tileentity/SignTileEntityRenderer.java
+@@ -118,7 +118,19 @@
+             if (p_217658_1_ != Blocks.field_222386_bZ && p_217658_1_ != Blocks.field_222394_cj) {
+                if (p_217658_1_ != Blocks.field_222389_ca && p_217658_1_ != Blocks.field_222395_ck) {
+                   if (p_217658_1_ != Blocks.field_222390_cb && p_217658_1_ != Blocks.field_222396_cl) {
+-                     return p_217658_1_ != Blocks.field_222391_cc && p_217658_1_ != Blocks.field_222397_cm ? field_217659_c : field_217664_h;
++                     if (p_217658_1_ != Blocks.field_222391_cc && p_217658_1_ != Blocks.field_222397_cm) {
++                        //Forge: don't forget to add your signs to valid blocks for entity: TileEntityType.SIGN.addValidBlocks(...)
++                        String path = p_217658_1_.getRegistryName().func_110623_a();
++                        if (path.endsWith("_wall_sign")) {
++                           return new ResourceLocation(p_217658_1_.getRegistryName().func_110624_b(), "textures/entity/signs/" + path.substring(0, path.length() - 10) + ".png");
```

Comment:
This code is unnecessarily duplicated, some simple regex would make it much cleaner i.e. `replaceAll("(_wall)?_sign", "")`

Reply:
Your are right, I will fix this. :)

Reply:
Alternatively just allow the sign tile entity to provide a resource location instead of auto-generation

Reply:
That is also a possibility, but the resource location of textures should lie inside the renderer. For armor items it is also generated inside the renderer.

Reply:
I added a new commit with the changes.